controllers/provider-azure/pkg/controller/infrastructure/actuator_delete.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infrastructure
+
+import (
+	"context"
+	"time"
+
+	"github.com/gardener/gardener-extensions/controllers/provider-azure/pkg/internal"
+	"github.com/gardener/gardener-extensions/controllers/provider-azure/pkg/internal/infrastructure"
+	"github.com/gardener/gardener-extensions/pkg/controller"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/utils/flow"
+)
+
+// Delete implements infrastructure.Actuator.
+func (a *actuator) Delete(ctx context.Context, infra *extensionsv1alpha1.Infrastructure, cluster *controller.Cluster) error {
+	clientAuth, err := internal.GetClientAuthData(ctx, a.client, infra.Spec.SecretRef.Namespace, infra.Spec.SecretRef.Name)
+	if err != nil {
+		return err
+	}
+
+	tf, err := internal.NewTerraformer(a.restConfig, clientAuth, infrastructure.TerraformerPurpose, infra.Namespace, infra.Name)
+	if err != nil {
+		return err
+	}
+
+	configExists, err := tf.ConfigExists()
```

Comment:
All this complexity with flow etc. is not needed for Azure. Just call `tf.Destroy` ([as it was in the Azure Botanist](https://github.com/gardener/gardener/blob/master/pkg/operation/cloudbotanist/azurebotanist/infrastructure.go#L76-L78)).

Reply:
I thought maybe in the future we would have to deleted certain infrastructure components like we need in other infrastructure (for cleanup purposes), so thought wouldn't hurt to add the flow now.